skcosmo/feature_selection/_greedy.py

```diff
@@ -0,0 +1,238 @@
+
+    Parameters
+    ----------
+
+    n_features_to_select : int or float, default=None
+        The number of features to select. If `None`, half of the features are
+        selected. If integer, the parameter is the absolute number of features
+        to select. If float between 0 and 1, it is the fraction of features to
+        select.
+
+    score_thresh_to_select : float, default=None
+        Threshold for the score. If `None` selection will continue until the
+        number or fraction given by n_features_to_select is chosen. Otherwise
+        will stop when the score falls below the threshold.
+
+    scoring : str, callable, list/tuple or dict, default=None
+        A single str (see :ref:`scoring_parameter`) or a callable
+        (see :ref:`scoring`) to evaluate the features. It is assumed that the
+        next feature to select is that which maximizes the score.
+
+        NOTE that when using custom scorers, each scorer should return a single
+        value. Metric functions returning a list/array of values can be wrapped
+        into multiple scorers that return one value each.
+
+    full : boolean
+        In the case that all non-redundant features are exhausted, choose
+        randomly from the remaining features.
+
+    progress_bar: boolean, default=False
+                  option to use `tqdm <https://tqdm.github.io/>`_
+                  progress bar to monitor selections
+
+    Attributes
+    ----------
+
+    n_features_to_select : int
+        The number of features that were selected.
+
+    X_selected_ : ndarray (n_samples, n_features_to_select)
+                  The features selected
+
+    selected_idx_ : ndarray of integers
+                    indices of the selected features, with respect to the
+                    original fitted matrix
+
+    support_ : ndarray of shape (n_features,), dtype=bool
+        The mask of selected features.
+
+    """
+
+    def __init__(
+        self,
+        scoring,
+        n_features_to_select=None,
+        score_thresh_to_select=None,
+        progress_bar=False,
+        full=False,
+    ):
+
+        self.n_features_to_select = n_features_to_select
+        self.n_selected_ = 0
+        self.scoring = scoring
+
+        if full and score_thresh_to_select is not None:
+            raise ValueError(
+                "You cannot specify both `score_thresh_to_select` and `full=True`."
+            )
+
+        self.score_threshold = score_thresh_to_select
+        self.full = full
+
+        if progress_bar:
+            self.report_progress = get_progress_bar()
+        else:
+            self.report_progress = lambda x: x
+
+    def fit(self, X, y=None, warm_start=False):
+        """Learn the features to select.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training vectors.
+        y : array-like of shape (n_samples,)
+            Target values.
+        warm_start : bool
```

Comment:
naming: `warm_start` => `continue_selection`

Reply:
This is an inherited naming convention from scikit-learn